inference-engine/tests/unit/cpu/mkldnn_memory_desc_test.cpp

```diff
@@ -45,6 +45,1133 @@ TEST(MemDescTest, Conversion) {
+            { dnnl::memory::format_tag::NChw16n16c,  Shape(SizeVector {4, 2, 10, 7 }) },  // blocked for 2 dims
+            { dnnl::memory::format_tag::Acdb16a,     Shape(SizeVector {96, 1, 7, 7 }) },  // same strides but not default order
+    };
+
+    for (const auto &p : payload) {
+        convertToTensorDesc(p.first, p.second);
+    }
+    ASSERT_ANY_THROW(convertToTensorDesc(dnnl::memory::format_tag::nChw16c, dynamicShape));
+}
+
+TEST(MemDescTest, isCompatibleTrueTests) {
+    ngraph::PartialShape UndefShape({{16}, {7, 15}, {-1, -1}, {-1, -1}});
+    Shape dynamicShape(UndefShape);
+
+    auto isCompatibleDnnlMemoryDesc = [] (dnnl::memory::format_tag fmt, const Shape& shape) {
+        if (shape.isStatic()) {
+            dnnl::memory::desc orig_desc{MKLDNNExtensionUtils::convertToDnnlDims(shape.getDims()), dnnl::memory::data_type::u8, fmt};
+            DnnlMemoryDescPtr dnnl_desc_ptr = MKLDNNExtensionUtils::makeDescriptor(orig_desc);
+            DnnlBlockedMemoryDesc dnnl_blk_desc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(*dnnl_desc_ptr);
+            BlockedMemoryDescPtr blk_desc_ptr = MemoryDescUtils::convertToBlockedMemoryDesc(dnnl_blk_desc.clone());
+            MemoryDescPtr cpu_blk_desc_ptr = std::make_shared<CpuBlockedMemoryDesc>(blk_desc_ptr->getPrecision(),
+                                                                                    blk_desc_ptr->getShape(),
+                                                                                    blk_desc_ptr->getBlockDims(),
+                                                                                    blk_desc_ptr->getOrder(),
+                                                                                    blk_desc_ptr->getOffsetPadding(),
+                                                                                    blk_desc_ptr->getOffsetPaddingToData(),
+                                                                                    blk_desc_ptr->getStrides());
+            auto dnnl_desc_ptr_copy = dnnl_desc_ptr->clone();
+            ASSERT_TRUE(dnnl_desc_ptr->isCompatible(*dnnl_desc_ptr) && dnnl_desc_ptr->isCompatible(dnnl_blk_desc) &&
+                        dnnl_desc_ptr->isCompatible(*blk_desc_ptr) && dnnl_desc_ptr->isCompatible(*cpu_blk_desc_ptr) &&
+                        dnnl_desc_ptr->isCompatible(*dnnl_desc_ptr_copy));
+        } else {
+            dnnl::memory::desc orig_desc{MKLDNNExtensionUtils::convertToDnnlDims(shape.getDims()), dnnl::memory::data_type::u8, fmt};
+            ASSERT_ANY_THROW(DnnlMemoryDescPtr dnnl_desc_ptr = MKLDNNExtensionUtils::makeDescriptor(orig_desc));
+        }
+    };
+    auto isCompatibleDnnlBlockedMemoryDesc = [] (dnnl::memory::format_tag fmt, const Shape& shape) {
+        DnnlBlockedMemoryDesc dnnl_blk_desc(shape, dnnl::memory::data_type::u8, fmt);
+        BlockedMemoryDescPtr blk_desc_ptr = MemoryDescUtils::convertToBlockedMemoryDesc(dnnl_blk_desc.clone());
```

Comment:
why do you need this?

you should also check that `isCompatible` for `DnnlMemoryDesc` return false
same for below